pkg/controller/operators/openshift/helpers.go

```diff
@@ -189,6 +193,22 @@ func desiredRelease(ctx context.Context, cli client.Client) (*semver.Version, er
 	return &desired, nil
 }
 
+func nextY(v semver.Version) (semver.Version, error) {
+	v.Build = nil // Builds are irrelevant
+
+	if len(v.Pre) > 0 {
+		// Dropping pre-releases is equivalent to incrementing Y
+		v.Pre = nil
+		v.Patch = 0
+
+		return v, nil
+	}
+
+	err := v.IncrementMinor() // Sets Y=Y+1 and Z=0
+
+	return v, err
```

Comment:
nit/not blocking: any reason we need to capture the error as a variable here instead of just returning the value of v.IncrementMinor() explicitly here?

Reply:
was just unsure about copy ordering in a return statement -- looks like it doesn't happen until all the expressions are evaluated: https://play.golang.org/p/tKT2N-UCMXz